protocol/messenger_profile_showcase.go

```diff
@@ -107,6 +109,46 @@ func (m *Messenger) validateCollectiblesOwnership(accounts []*identity.ProfileSh
 	return nil
 }
 
+func (m *Messenger) validateCommunitiesMembership(communities []*identity.ProfileShowcaseCommunity, contactPubKey *ecdsa.PublicKey) ([]*identity.ProfileShowcaseCommunity, error) {
+	validatedCommunities := []*identity.ProfileShowcaseCommunity{}
+
+	for _, communityEntry := range communities {
+		community, err := m.FetchCommunity(&FetchCommunityRequest{
+			CommunityKey:    communityEntry.CommunityID,
+			Shard:           nil,
+			TryDatabase:     true,
+			WaitForResponse: true,
+		})
+		if err != nil {
+			m.logger.Warn("failed to fetch community for profile entry ", zap.Error(err))
+		}
+
+		if community != nil && community.Encrypted() {
+			grant, err := community.VerifyGrantSignature(communityEntry.Grant)
+			if err != nil {
+				m.logger.Warn("failed to verify grant signature ", zap.Error(err))
+				communityEntry.MembershipStatus = identity.ProfileShowcaseMembershipStatusNotAMember
+			} else {
+				if grant != nil && bytes.Equal(grant.MemberId, crypto.CompressPubkey(contactPubKey)) {
+					communityEntry.MembershipStatus = identity.ProfileShowcaseMembershipStatusProvenMember
+				} else { // Show as not a member if membership can't be proven
+					communityEntry.MembershipStatus = identity.ProfileShowcaseMembershipStatusNotAMember
+				}
+			}
+		} else if community != nil {
+			// Use member list as a proof for unecrypted communities
+			if community.HasMember(contactPubKey) {
+				communityEntry.MembershipStatus = identity.ProfileShowcaseMembershipStatusProvenMember
+			} else {
+				communityEntry.MembershipStatus = identity.ProfileShowcaseMembershipStatusNotAMember
+			}
+		}
```

Comment:
Should we mark `MembershipStatus` as `ProfileShowcaseMembershipStatusNotAMember` in the `else` here?

Or maybe we just shouldn't append? We can `continue` in the `else` then.

Reply:
it's defaulted as `ProfileShowcaseMembershipStatusUnproven` in persistence [here](https://github.com/status-im/status-go/pull/4982/files#diff-32b4e8b9487c63be352bc42c70435075213df4e163941b52aadcac0e9d4c85d9R409). I'll do it in more obvious way